includes/admin/class-amp-post-meta-box.php

```diff
@@ -151,17 +151,20 @@ public function enqueue_admin_assets() {
 
 		wp_styles()->add_data( self::ASSETS_HANDLE, 'rtl', 'replace' );
 
-		$script_deps_path    = AMP__DIR__ . '/assets/js/' . self::ASSETS_HANDLE . '.deps.json';
-		$script_dependencies = file_exists( $script_deps_path )
-			? json_decode( file_get_contents( $script_deps_path ), false ) // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		$asset_file = AMP__DIR__ . '/assets/js/' . self::ASSETS_HANDLE . '.asset.php';
+		$asset      = file_exists( $asset_file )
```

Comment:
The `file_exists()` check here could cause this issue to raise its head again: https://github.com/ampproject/amp-wp/issues/2670

Perhaps what is needed is something like https://github.com/ampproject/amp-wp/issues/2670 where the build step ensures that the file gets loaded unconditionally if running in a plugin build?

Reply:
That sounds like the `file_exists()` check is not needed at all, since the plugin fails way earlier when running from source.